docs/cloud/features/scheduler/dagster.md

```diff
@@ -1,3 +1,334 @@
+
+This means that if you run a plan that adds or removes models from your Tobiko Cloud project, they will not show in Dagster by default.
+
+### Automatic method
+
+Dagster runs user code in an isolated sandbox. This means that the user code cannot easily communicate with the Dagster Webserver to trigger a reload of the Asset definitions.
+
+However, it is possible to connect to the Dagster GraphQL API from within user code to trigger the reload of the Code Location containing the Tobiko Cloud asset definitions.
+
+This is enabled by default if you specify the following extra parameters when instantiating `SQLMeshEnterpriseDagster`:
+
+```python title="definitions.py" linenums="4-5"
+sqlmesh = SQLMeshEnterpriseDagster(
+    url=EnvVar("TOBIKO_CLOUD_BASE_URL").get_value(),
+    token=EnvVar("TOBIKO_CLOUD_TOKEN").get_value(),
+    dagster_graphql_host="localhost",
+    dagster_graphql_port=3000
+)
+```
+
+The `dagster_graphql_host` and `dagster_graphql_port` are deployment-specific. This should be the hostname/port that the Dagster GraphQL endpoint is available in.
+
+In the case of local development when Dagster is started via `dagster dev`, this is typically `localhost:3000`.
+
+When the Dagster GraphQL server location is known, the Sensor that triggers the `mirror` job will first issue a GraphQL request to reload the Code Location. This will pick up any new Asset and Asset Check definitions.
+
+Once the Code Location is reloaded, the `mirror` job executes as usual with the updated definitions.
+
+### Manual method
+
+At any time, you can click the "Reload" button against the Code Location in Dagster. This will pull through any new Asset and Asset Check definitions from Tobiko Cloud.
+
+![Dagster reload code location](./dagster/reload_code_location.png)
+
+## Attaching custom logic
+
+Dagster includes a robust events system. You can listen for events on any of the Assets or Jobs exposed by our Dagster integration and respond to them by running your own custom logic.
+
+To listen for materialization events on Assets, you can use an [Asset Sensor](https://docs.dagster.io/concepts/partitions-schedules-sensors/asset-sensors).
+
+To listen for job runs, you can use a [Run Status Sensor](https://docs.dagster.io/concepts/partitions-schedules-sensors/sensors#run-status-sensors).
+
+Dagster also provides a framework called [Declarative Automation](https://docs.dagster.io/concepts/automation/declarative-automation) that builds on top of these.
+
+### Examples
+
+Here are some examples of running custom logic in response to Tobiko Cloud events.
+
+Note that Dagster has a lot of flexibility in how it can be configured and the methods we describe below arent necessarily the right choice for every configuration. We recommend familiarizing yourself with Dagster's [Automation](https://docs.dagster.io/concepts/automation) features to get the most out of your Tobiko Cloud deployment with Dagster.
```

Comment:
```suggestion
Note that Dagster has a lot of flexibility in how it can be configured and the methods we describe below aren't necessarily the right choice for every configuration. We recommend familiarizing yourself with Dagster's [Automation](https://docs.dagster.io/concepts/automation) features to get the most out of your Tobiko Cloud deployment with Dagster.
```